src/main/cpp/src/get_json_object.cu

```diff
@@ -74,10 +74,37 @@ struct path_instruction {
 };
 
 // TODO parse JSON path
-thrust::optional<rmm::device_uvector<path_instruction>> parse_path(
-  cudf::string_scalar const& json_path)
+rmm::device_uvector<path_instruction> parse_path(std::vector<int32_t> const& path_types,
+                                                 std::vector<std::string> const& path_names,
+                                                 std::vector<int64_t> const& path_indexes,
+                                                 rmm::cuda_stream_view stream)
 {
-  return thrust::nullopt;
+  std::vector<path_instruction> path_instructions;
+  for (size_t i = 0; i < path_types.size(); i++) {
+    switch (path_types[i]) {
+      case 0:  // subscript
+        path_instructions.push_back(path_instruction(path_instruction_type::subscript));
+        break;
+      case 1:  // wildcard
+        path_instructions.push_back(path_instruction(path_instruction_type::wildcard));
+        break;
+      case 2:  // key
+        path_instructions.push_back(path_instruction(path_instruction_type::key));
+        break;
+      case 3:  // index
+        path_instructions.push_back(path_instruction(path_instruction_type::index));
+        path_instructions.back().index = path_indexes[i];
+        break;
+      case 4:  // named
+        path_instructions.push_back(path_instruction(path_instruction_type::named));
+        path_instructions.back().name =
+          cudf::string_view(path_names[i].c_str(), path_names[i].size());
+        break;
+      default: throw std::runtime_error("Invalid path type");
+    }
+  }
+  return cudf::detail::make_device_uvector_sync(
+    path_instructions, stream, rmm::mr::get_current_device_resource());
```

Comment:
Oh this will crash. The `cudf::string_view` objects are viewing host strings. We cannot process in device using view of the host strings.

In general, we only create `string_view` for local processing and should never pass it across host<=>device. Depending on the size of `path_names`, we may want to create an `std::vector` of `string_scalar` or a strings column and use it instead.